tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/OpenIdTests.java

```diff
@@ -0,0 +1,210 @@
+
+            UserRepresentation user = new UserRepresentation();
+            user.setEnabled(true);
+            user.setFirstName(firstName);
+            user.setLastName(lastName);
+            user.setUsername(userName);
+            user.setEmail(email);
+
+            userId = CreatedResponseUtil.getCreatedId(keycloak.realm("jetty").users().create(user));
+
+            CredentialRepresentation passwordCred = new CredentialRepresentation();
+            passwordCred.setTemporary(false);
+            passwordCred.setType(CredentialRepresentation.PASSWORD);
+            passwordCred.setValue(password);
+
+            // Set password credential
+            keycloak.realm("jetty").users().get(userId).resetPassword(passwordCred);
+        }
+    }
+
+    @AfterAll
+    public static void stopKeycloak()
+    {
+        if (KEYCLOAK_CONTAINER.isRunning())
+        {
+            KEYCLOAK_CONTAINER.stop();
+        }
+    }
+
+    public static Stream<Arguments> tests()
+    {
+        return Stream.of(
+                Arguments.of("ee9", "ee9-openid"),
+                Arguments.of("ee10", "openid")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("tests")
+    public void testOpenID(String env, String openIdModule) throws Exception
+    {
+        Path jettyBase = newTestJettyBaseDirectory();
+        String jettyVersion = System.getProperty("jettyVersion");
+        JettyHomeTester distribution = JettyHomeTester.Builder.newInstance()
+                .jettyVersion(jettyVersion)
+                .jettyBase(jettyBase)
+                .build();
+
+        String[] args1 = {
+                "--create-startd",
+                "--approve-all-licenses",
+                "--add-to-start=http," + toEnvironment("webapp", env) + "," + toEnvironment("deploy", env) + "," + openIdModule
+        };
+
+        try (JettyHomeTester.Run run1 = distribution.start(args1))
+        {
+            assertTrue(run1.awaitFor(START_TIMEOUT, TimeUnit.SECONDS));
+            assertEquals(0, run1.getExitValue());
+
+            Path webApp = distribution.resolveArtifact("org.eclipse.jetty." + env + ":jetty-" + env + "-test-openid-webapp:war:" + jettyVersion);
+            distribution.installWar(webApp, "test");
+            String openIdProvider = KEYCLOAK_CONTAINER.getAuthServerUrl() + "/realms/jetty";
+            LOGGER.info("openIdProvider: {}", openIdProvider);
+
+            int port = Tester.freePort();
+            String[] args2 = {
+                    "jetty.http.port=" + port,
+                    "jetty.ssl.port=" + port,
+                    "jetty.openid.provider=" + openIdProvider,
+                    "jetty.openid.clientId=" + clientId,
+                    "jetty.openid.clientSecret=" + clientSecret,
+                    //"jetty.server.dumpAfterStart=true",
+            };
+
+            try (JettyHomeTester.Run run2 = distribution.start(args2); WebClient webClient = new WebClient();)
```

Comment:
I would have preferred use of the jetty `HttpClient` instead of the gargoylesoftware one.

Reply:
bear in mind this is using now a real opened provider so there is a redirection to web page for login. 
And not sure we really want to spend time parsing the returned HTML to find all the information such: cookie, all ids pass, session etc....
it looks the easiest way to mimic a browser.

Reply:
The Jetty client will handle the session cookies, and can follow redirects, and here it is always reading the full content as a string and doing `containsString(...)` checks on it, so no HTML parsing needed. And then there is a FORM submission which is also easy to do with the Jetty HttpClient.